x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_api/generate_engine_api_key_modal/generate_engine_api_key_modal.tsx

```diff
@@ -0,0 +1,196 @@
+                            {
+                              defaultMessage: 'Generate Read only key',
+                            }
+                          )}
+                        </EuiButton>
+                      </EuiFlexItem>
+                    </>
+                  ) : (
+                    <EuiFlexItem>
+                      <EuiFormLabel>{keyName}</EuiFormLabel>
+                      <EuiSpacer size="xs" />
+                      <EuiFlexGroup alignItems="center">
+                        <EuiFlexItem>
+                          <EuiCodeBlock
+                            aria-label={keyName}
+                            fontSize="m"
+                            paddingSize="m"
+                            color="dark"
+                            isCopyable
+                          >
+                            {apiKey}
+                          </EuiCodeBlock>
+                        </EuiFlexItem>
+                        <EuiFlexItem grow={false}>
+                          <EuiButtonIcon
+                            data-telemetry-id={`entSearchContent-engines-api-generateEngineApiKeyModal-csvDownloadButton`}
+                            aria-label={i18n.translate(
+                              'xpack.enterpriseSearch.content.engine.api.generateEngineApiKeyModal.csvDownloadButton',
+                              { defaultMessage: 'Download API key' }
+                            )}
+                            iconType="download"
+                            href={encodeURI(`data:text/csv;charset=utf-8,${apiKey}`)}
+                            download={`${keyName}.csv`}
+                          />
+                        </EuiFlexItem>
+                      </EuiFlexGroup>
+                    </EuiFlexItem>
+                  )}
+                </EuiFlexGroup>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiFlexGroup direction="row">
+                  <EuiFlexItem>
+                    <EuiText size="s" color="#006bb8">
+                      <p>
+                        {i18n.translate(
+                          'xpack.enterpriseSearch.content.engine.api.generateEngineApiKeyModal.apiKeyWarning',
+                          {
+                            defaultMessage:
+                              "Elastic does not store API keys. Once generated, you'll only be able to view the key one time. Make sure you save it somewhere secure. If you lose access to it you'll need to generate a new API key from this screen.",
+                          }
+                        )}
+                      </p>
+                    </EuiText>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPanel>
+        </>
+      </EuiModalBody>
+      <EuiModalFooter>
+        {apiKey ? (
+          <EuiButton
+            data-telemetry-id={`entSearchContent-engines-api-generateEngineApiKeyModal-done`}
```

Comment:
nitpick: can just be a string (I'm gonna stop noting these after this one, but there's more)

Reply:
yep thats me c&p :joy: Ill find them all. thanks!